server/src/main/java/org/elasticsearch/index/search/MatchQuery.java

```diff
@@ -320,11 +320,11 @@ private boolean hasPositions(MappedFieldType fieldType) {
             super(analyzer);
             this.fieldType = fieldType;
             if (hasPositions(fieldType)) {
-                setAutoGenerateMultiTermSynonymsPhraseQuery(autoGenerateSynonymsPhraseQuery);
+                setAutoGenerateMultiTermSynonymsPhraseQuery(MatchQuery.this.autoGenerateSynonymsPhraseQuery);
```

Comment:
To me it'd be clearer to pass these two flags through the constructor, as it avoids having to figure out the inheritance structure.